src/kernel_serial.rs

```diff
@@ -1,27 +1,19 @@
 use core::fmt::{self, Write};
 
-use crate::drivers::ns16550::{Ns16550, QEMU_VIRT_BASE_ADDRESS};
-
-static mut KERNEL_SERIAL: Option<Ns16550> = None;
-
-fn check_init() {
-    unsafe {
-        match &KERNEL_SERIAL {
-            None => {
-                KERNEL_SERIAL = Some(Ns16550::new(QEMU_VIRT_BASE_ADDRESS));
-            }
-            Some(_) => {}
-        }
-    }
+pub trait Console {
```

Comment:
Since the filename is `kernel_serial`, maybe this should be called `Serial`?

Reply:
No

Reply:
`kernel_console.rs` it is then 👍

Reply:
renamed